src/components/UI/Text/TimeStampInput.vue

```diff
@@ -0,0 +1,142 @@
+<template>
+  <div>
+    <div>
+      <p class="text-xs pl-2">{{ title }}</p>
+    </div>
+    <div class="flex flex-row space-x-1">
+      <!-- hour input -->
+      <input-text
+        v-if="config.showHour"
+        :placeholder="'HH'"
+        v-model:value="hour"
+        class="w-12"
+        :type="{ boxType: 'input', inputType: 'number' }"
+        :min="0"
+        :max="23"
+      ></input-text>
+
+      <p class="self-center" v-if="config.showHour && config.showMinute">:</p>
+
+      <!-- minute input -->
+      <input-text
+        v-if="config.showMinute"
+        :placeholder="'MM'"
+        v-model:value="minute"
+        class="w-12"
+        :type="{ boxType: 'input', inputType: 'number' }"
+        :min="0"
+        :max="59"
+      ></input-text>
+
+      <p class="self-center" v-if="config.showMinute && config.showSecond">:</p>
+
+      <!-- second input -->
+      <input-text
+        v-if="config.showSecond"
+        :placeholder="'SS'"
+        v-model:value="second"
+        class="w-12"
+        :type="{ boxType: 'input', inputType: 'number' }"
+        :min="0"
+        :max="59"
+      ></input-text>
+
+      <p class="self-center" v-if="config.showSecond && config.showMillisecond">:</p>
+
+      <!-- millisecond input -->
+      <input-text
+        v-if="config.showMillisecond"
+        :placeholder="'mmm'"
+        v-model:value="millisecond"
+        class="w-16"
+        :type="{ boxType: 'input', inputType: 'number' }"
+        :min="0"
+        :max="999"
+      ></input-text>
+    </div>
+  </div>
+</template>
+
+<script>
+import InputText from "@/components/UI/Text/InputText.vue";
+
+export default {
+  components: {
+    InputText,
+  },
+  props: {
+    config: {
+      // which box to show can be controled
+      // by default - all 4 boxes will show
+      default: function () {
+        return {
+          showHour: true,
+          showMinute: true,
+          showSecond: true,
+          showMillisecond: true,
+        };
```

Comment:
what about the case when someone passes config with only one of the 4? what is the value for the remaining then?